File: 15. UnSharp_HighBoost_Sobel Filtering/Sobel.py
import cv2
import numpy as np

img = cv2.imread('house.tif', 0)
cv2.imshow("Original", img)

x = cv2.Sobel(img, -1, 0, 1, ksize=3)
y = cv2.Sobel(img, -1, 1, 1, ksize=3)
cv2.imshow("X", x)
cv2.imshow("Y", y)

x = np.array(x, np.float64)
y = np.array(y, np.float64)
z = np.sqrt(x*x + y*y)
z = np.array(z, np.uint8)
cv2.imshow("Sobel", z)

File: 15. UnSharp_HighBoost_Sobel Filtering/Sobel.m
x = fspecial('sobel');
y = x';
img = imread('house.tif');
imgx = imfilter(img, x);
imgy = imfilter(img, y);
x1 = double(imgx);
y1 = double(imgy);
z = sqrt(x1.*x1 + y1.*y1);
z=uint8(z);

figure('Name', 'Original');
imshow(img);
figure('Name', 'Horizontal');
imshow(imgx);
figure('Name', 'Vertical');
imshow(imgy);
figure('Name', 'Sobel');
imshow(z);

%Comparision with Laplacian Filter
lap = fspecial('laplacian');
imglap = imfilter(img, lap);
figure('Name', 'Laplacian Filter');
imshow(imglap, []);
